src/main/java/net/dries007/tfc/client/render/blockentity/TFCHangingSignBlockEntityRenderer.java

```diff
@@ -58,6 +71,42 @@ public void render(SignBlockEntity sign, float partialTick, PoseStack poseStack,
         final WoodType woodtype = SignBlock.getWoodType(signblock);
         final HangingSignRenderer.HangingSignModel model = this.hangingSignModels.get(woodtype);
         model.evaluateVisibleParts(blockstate);
-        ((SignRendererAccessor) (Object) this).invoke$renderSignWithText(sign, poseStack, buffer, light, overlay, blockstate, signblock, woodtype, model);
+        if(signblock instanceof ITFCHangingSignBlock tfcSignBlock) {
+            final Metal metal = Metal.MANAGER.getOrThrow(tfcSignBlock.metal());
+            this.renderSignWithText(sign, poseStack, buffer, light, overlay, blockstate, signblock, woodtype, metal, model);
```

Comment:
My intuition is the hanging sign block should not store a _metal_ but rather a texture or model location (ResourceLocation). This way we don't have to do a potentially dangerous lookup like this